oo/classical-inheritance-es5.md

```diff
@@ -0,0 +1,166 @@
+
+We then instantiate `Comment` and `Discussion` instances using the `new` keyword as we would do with ES6 classes.
+
+
+## Inheritance with factory functions 
+
+```js
+function Post (options) {
+  return {
+    author: options.author,
+    content: options.content,
+    createdAt: Date.now(),
+    getCreatedAt: function () {
+      return this.createdAt
+    }
+  }
+}
+
+function Comment (options) {
+  // call the 'parent' factory function 
+  var comment = Post(options) 
+
+  // mutate
+  comment.likeCount = 0
+  comment.like = function () {
+    this.likeCount ++
+  }
+  
+  return comment
+}
+
+function Discussion (options) {
+  var discussion = Post(options)  
+  discussion.title = options.title
+  discussion.comments = []
+
+  discussion.addComment = function (comment) {
+    this.comments.push(comment)
+  }
+
+  discussion.getCommentAtIndex = function (index) {
+    return this.comments[index]
+  }
+
+  return discussion
+}
+
+var oneRingDiscussion = Discussion({
+  title: 'Destroying the one ring',
+  content: "Let's brainstorm ideas for destroying the one ring",
+  author: 'Gandalf'
+})
+
+var frodoComment = Comment({ 
+  author: 'Frodo', 
+  content: 'One idea is to throw it into a volcano. What does everyone think?'
+})
+
+console.log(frodoComment.getCreatedAt())
+// 1471297024140 (unix time)
+oneRingDiscussion.addComment(frodoComment)
+console.log(frodoComment === oneRingDiscussion.getCommentAtIndex(0))
+// true
+
+console.log(frodoComment instanceof Comment)
+// false
+console.log(frodoComment instanceof Object)
+// true
+
+```
+
+To use factory functions we simply `return` plain objects. We may mutate these objects inside the factory function scope and add new properties.
+
+
+## Differences
+
+We call Constructor functions using the `new` keyword while we call factory functions like regular functions. Constructor function instances can be checked against their constructor with the `instanceof` keyword while factory function instances are regular `Object` instances.
+
+The more informative `instanceof` behaviour of constructor functions may be useful in codebases where checking the pseudo-class of instances is important for error catching, e.g.:
+
+```js
+Discussion.prototype.addComment = function (comment) {
+  if (!(comment instanceof Comment)) {
+    throw new Error("you've tried to add something that isn't a comment!")
+  }
+  this.comments.push(comment)
+}
+```
+
+Other developers may prefer the simplicity of factory functions.
+
```

Comment:
It's probably worth mentioning that the other difference with factory functions done this way is that it isn't using prototypal inheritance (the way JavaScript does OO). I think it's worth mentioning that factory functions are a super useful pattern in all types of software development (including JavaScript), but if they are expressing an inheritance relationship, it's a best practice to use prototypal inheritance rather than implement  something custom.